erigon-lib/state/aggregator.go

```diff
@@ -724,7 +720,16 @@ func (a *Aggregator) mergeLoopStep(ctx context.Context, toTxNum uint64) (somethi
 	return true, nil
 }
 
+// TODO: merge must have own semphore
 func (a *Aggregator) MergeLoop(ctx context.Context) error {
+	if dbg.NoMerge() || !a.mergingFiles.CompareAndSwap(false, true) {
+		return nil // currently merging or merge is prohibited
+	}
+
+	a.wg.Add(1)
```

Comment:
next pattern is race-condition: 
```
go func() {
  a.wg.Add(1)
  defer a.wg.Done()
}
a.wg.Wait()
```

Reply:
its a race because wait is straight after routine called and in this case wg could be negative (not yet added), we have wait only in Close(). To be honest i thought that this exact usage is not a race or no more race than previous usage when we had a.wg.Add(1) on routine one level above